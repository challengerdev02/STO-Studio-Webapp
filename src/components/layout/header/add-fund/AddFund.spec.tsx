import {
  cleanup,
  fireEvent,
  queryByAttribute,
  queryByRole,
  render,
  waitFor,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { AddFund } from './index';

jest.mock('qrcode');
jest.mock('@/shared/utils');

describe('AddFund', function () {
  afterEach(cleanup);
  let props = {
    visibility: true,
    address: 'This is a test title',
    onVisibilityChange: jest.fn(),
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<AddFund {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should display the modal when "visible" is true', async () => {
    // @ts-ignore
    setup();
    expect(queryByRole(document.body, 'dialog')).toBeInTheDocument();
  });

  it('should fire onCancel callback when modal is closed', async () => {
    // @ts-ignore
    const { container } = setup();

    await waitFor(() => {
      const btn = queryByAttribute('aria-label', document.body, 'Close');

      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'dialog')).toBeInTheDocument();
    });
  });

  it('should fire Copy Address callback address NEW', async () => {
    // @ts-ignore
    const { container } = setup();

    await waitFor(() => {
      const btn = queryByAttribute(
        'class',
        document.body,
        'ant-btn ant-btn-primary'
      );

      if (btn) {
        fireEvent.click(btn);
      }
    });

    await waitFor(() => {
      expect(queryByRole(document.body, 'dialog')).toBeInTheDocument();
    });
  });
});
