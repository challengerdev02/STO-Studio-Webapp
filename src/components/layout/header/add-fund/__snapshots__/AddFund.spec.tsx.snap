// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFund should match with snapshot 1`] = `<div />`;
